File: 1813-maximum-erasure-value/1813-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
         int n = nums.size();

        vector<int> cus(n, 0);
        cus[0] = nums[0];
        for(int i = 1; i < n; i++)
            cus[i] = cus[i-1] + nums[i];
        
        vector<int> mp(10001, -1);  
        int maxS   = 0;
        int i      = 0;
        int j      = 0;
        
        while(j < n) {
            i       = max(i, mp[nums[j]]+1);
            int jthSum  = cus[j];
            int ithSum  = i-1 < 0 ? 0 : cus[i-1];
            maxS        = max(maxS, jthSum - ithSum);
            mp[nums[j]] = j;

            j++;
        }
        
        return maxS;
    }
};

File: 1813-maximum-erasure-value/1813-maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        int r=0,l=0;
            Set<Integer>st=new HashSet<>();
            int sum=0;
            int max=0;
            while(r<nums.length)
            {
                int k=nums[r];
                    while(st.contains(k))
                    {
                        st.remove(nums[l]);
                        sum-=nums[l];
                        l++;
                    }
                    st.add(k);
                    sum += k;
                    max = Math.max(max, sum);
                    r++;
            }  
            return max;
    }
}